### PR TITLE
Merge the master branch into epic-es7

### PR DESCRIPTION
**Pull request by anatol-sialitski**

### CODE REVIEW

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 8
- Added 5
           

Complexity increasing per file
==============================
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/OsgiSupport.java  7
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/reader/ZipDumpReader.java  6
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/DeleteProjectRolesCommand.java  3
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ProjectAccessException.java  12
- modules/admin/admin-event/src/main/java/com/enonic/xp/admin/event/impl/EventEndpoint.java  5
- modules/core/core-blobstore/src/testFixtures/java/com/enonic/xp/internal/blobstore/MemoryBlobRecord.java  3
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/CreateOrModifyProjectParamsJson.java  1
- modules/core/core-api/src/main/java/com/enonic/xp/project/ProjectPermissions.java  1
- modules/admin/admin-event/src/main/java/com/enonic/xp/admin/event/impl/WebsocketManagerImpl.java  1
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/repository/UpdateRepositoryEntryParams.java  2
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/SingleTaskReporter.java  1
- modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationLoader.java  6
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ContentAuditLogSupportImpl.java  6
- modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationRegistryImpl.java  9
- modules/lib/lib-project/src/test/resources/test/CreateProjectHandlerTest.js  2
- modules/script/script-impl/src/main/java/com/enonic/xp/script/impl/async/ScriptAsyncServiceImpl.java  2
- modules/core/core-repo/src/testFixtures/java/com/enonic/xp/repo/impl/node/AbstractNodeTest.java  3
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ModifyPermissionsParamsJson.java  1
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/status/objects/TopicObjectReport.java  1
- modules/lib/lib-repo/src/main/java/com/enonic/xp/lib/repo/GetRepositoryBinaryHandler.java  1
- modules/lib/lib-project/src/main/resources/lib/xp/examples/project/create.js  1
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/init/ContentInitializer.java  2
- modules/launcher/launcher-impl/src/main/java/com/enonic/xp/launcher/impl/framework/FrameworkLifecycleActor.java  3
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectReadAccessJson.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/GetProjectLanguageCommand.java  1
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/AddProjectPermissionsHandler.java  1
- modules/core/core-elasticsearch/src/main/java/com/enonic/xp/elasticsearch/impl/ElasticsearchCluster.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/ApplyProjectReadAccessPermissionsCommand.java  3
- modules/core/core-api/src/main/java/com/enonic/xp/app/ApplicationBundleUtils.java  1
- modules/admin/admin-event/src/main/java/com/enonic/xp/admin/event/impl/WebsocketEventExecutorImpl.java  1
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/elasticsearch/snapshot/ClusterRestarter.java  3
- modules/lib/lib-project/src/test/java/com/enonic/xp/lib/project/ModifyProjectHandlerTest.java  1
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/CreateProjectRootAccessListCommand.java  3
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/HazelcastSessionStoreFactoryActivator.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/content/task/ApplyPermissionsTaskMessageGenerator.java  2
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/content/CmsResourceDynamicFeature.java  2
- modules/lib/lib-project/src/test/java/com/enonic/xp/lib/project/CreateProjectHandlerTest.java  1
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/NullSystemDumpListener.java  1
- modules/lib/lib-auth/src/main/resources/lib/xp/auth.js  1
- modules/lib/lib-project/src/main/resources/lib/xp/examples/project/delete.js  1
- modules/lib/lib-content/src/main/java/com/enonic/xp/lib/content/mapper/ContentsResultMapper.java  2
- modules/core/core-hazelcast/src/test/java/com/enonic/xp/core/impl/hazelcast/HazelcastConfigServiceImplTest.java  1
- modules/launcher/launcher-impl/src/main/java/com/enonic/xp/launcher/impl/framework/FrameworkLifecycleService.java  1
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/blobstore/ZipDumpReadBlobStore.java  2
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/UpdateProjectRolesCommand.java  1
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/HazelcastActivator.java  2
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/writer/ZipDumpWriter.java  3
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/HazelcastConfigServiceImpl.java  4
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/WebSessionStoreConfigServiceImpl.java  4
- modules/lib/lib-repo/src/main/resources/lib/xp/examples/repo/modifyRemoveProperty.js  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectsJson.java  1
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/status/objects/HazelcastObjectsReporter.java  7
- modules/lib/lib-project/src/test/java/com/enonic/xp/lib/project/GetProjectHandlerTest.java  1
- modules/lib/lib-project/src/test/resources/test/GetProjectHandlerTest.js  2
- modules/web/web-session/src/test/java/com/enonic/xp/web/session/impl/WebSessionStoreConfigServiceImplTest.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/ApplyProjectLanguageCommand.java  1
- modules/core/core-api/src/main/java/com/enonic/xp/content/FindContentIdsByQueryResult.java  1
- modules/admin/admin-event/src/main/java/com/enonic/xp/admin/event/impl/WebsocketEventListener.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/ProjectReadAccess.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/GetProjectReadAccessCommand.java  3
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectPermissionsJson.java  1
- modules/core/core-api/src/main/java/com/enonic/xp/project/ModifyProjectParams.java  1
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/command/GetProjectReadAccessCommand.java  1
- modules/web/web-jetty/src/testFixtures/java/com/enonic/xp/web/jetty/impl/JettyTestSupport.java  1
- modules/core/core-hazelcast/src/test/java/com/enonic/xp/core/impl/hazelcast/status/cluster/HazelcastClusterReporterTest.java  1
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/NullSystemLoadListener.java  1
- modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationServiceImpl.java  3
- modules/lib/lib-project/src/test/java/com/enonic/xp/lib/project/ListProjectsHandlerTest.java  1
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/BaseProjectHandler.java  1
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/status/cluster/HazelcastMemberState.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/content/json/AccessControlEntriesJson.java  3
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/GetProjectRolesCommand.java  1
- modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/rendering/RendererDelegateImpl.java  2
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/page/GetPageTemplateBySiteCommand.java  3
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/AbstractProjectRootCommand.java  1
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/AbstractProjectRolesCommand.java  7
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/writer/AbstractDumpWriter.java  7
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ProjectNotFoundException.java  1
- modules/script/script-impl/src/main/java/com/enonic/xp/script/impl/executor/ScriptExecutorFactory.java  5
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/status/objects/HazelcastObjectsReport.java  1
- modules/lib/lib-project/src/test/java/com/enonic/xp/lib/project/ModifyProjectPermissionsHandlerTest.java  1
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/CreateProjectHandler.java  5
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ProjectPermissionsContextManagerImpl.java  6
- modules/lib/lib-project/src/test/java/com/enonic/xp/lib/project/DeleteProjectHandlerTest.java  1
- modules/admin/admin-event/src/main/java/com/enonic/xp/admin/event/impl/EventEndpointFactory.java  1
- modules/lib/lib-project/src/test/resources/test/ModifyProjectHandlerTest.js  1
- modules/core/core-api/src/testFixtures/java/com/enonic/xp/support/AbstractEqualsTest.java  2
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/status/cluster/HazelcastClusterReporter.java  2
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/mapper/ProjectMapper.java  3
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/UpdateProjectRoleNamesCommand.java  2
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/mapper/ProjectPermissionsMapper.java  2
- modules/core/core-api/src/main/java/com/enonic/xp/project/ModifyProjectIconParams.java  1
- modules/core/core-api/src/main/java/com/enonic/xp/repository/UpdateRepositoryParams.java  1
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ContentAuditLogExecutorImpl.java  1
- modules/core/core-api/src/main/java/com/enonic/xp/sortvalues/SortValuesProperty.java  3
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ModifyLanguageParamsJson.java  1
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/RemoveProjectPermissionsHandler.java  1
- modules/core/core-api/src/main/java/com/enonic/xp/project/Projects.java  2
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/init/IssueInitializer.java  1
- modules/core/core-api/src/main/java/com/enonic/xp/node/NodeBranchEntries.java  1
- modules/lib/lib-project/src/main/resources/lib/xp/project.js  2
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ProjectAccessHelper.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/DeleteProjectParamsJson.java  1
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/status/objects/QueueObjectReport.java  1
- modules/web/web-session/src/test/java/com/enonic/xp/web/session/impl/NullSessionStoreFactoryActivatorTest.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/ClusteredTaskManagerImpl.java  6
- modules/core/core-api/src/main/java/com/enonic/xp/repository/EditableRepository.java  1
- modules/lib/lib-project/src/test/resources/test/ListProjectsHandlerTest.js  1
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/ModifyProjectPermissionsHandler.java  3
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ProjectServiceImpl.java  5
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/reader/AbstractDumpReader.java  4
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/ModifyProjectHandler.java  5
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/DeleteProjectHandler.java  1
- modules/core/core-api/src/main/java/com/enonic/xp/project/CreateProjectParams.java  1
- modules/lib/lib-project/src/test/resources/test/DeleteProjectHandlerTest.js  2
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ProjectAccessRequiredException.java  1
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/blobstore/ZipDumpWriteBlobStore.java  4
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/CreateProjectRolesCommand.java  2
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/TasksReporterCallable.java  1
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/DumpServiceImpl.java  1
- modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/config/ApplicationConfigReloader.java  1
- modules/core/core-api/src/main/java/com/enonic/xp/project/Project.java  8
- modules/core/core-api/src/main/java/com/enonic/xp/project/ProjectName.java  2
- modules/web/web-jetty/src/main/java/com/enonic/xp/web/jetty/impl/session/JettySessionStoreConfiguratorImpl.java  1
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/status/cluster/HazelcastClusterReport.java  1
- modules/core/core-app/src/testFixtures/java/com/enonic/xp/core/impl/app/BundleBasedTest.java  2
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/elasticsearch/result/SortValuesPropertyFactory.java  1
- modules/core/core-image/src/main/java/com/enonic/xp/core/impl/image/ImageServiceImpl.java  2
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/AbstractSessionStoreFactoryActivator.java  3
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/command/GetProjectLanguageCommand.java  1
- modules/lib/lib-repo/src/main/resources/lib/xp/examples/repo/modifyScoped.js  1
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/PathRef.java  5
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ModifyReadAccessParamsJson.java  1
- modules/lib/lib-project/src/test/java/com/enonic/xp/lib/project/ModifyProjectReadAccessHandlerTest.java  1
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/blobstore/DumpBlobRecord.java  1
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/mapper/ProjectReadAccessMapper.java  2
- modules/core/core-blobstore/src/testFixtures/java/com/enonic/xp/internal/blobstore/MemoryBlobStore.java  2
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/AbstractProjectCommand.java  1
- modules/script/script-impl/src/main/java/com/enonic/xp/script/impl/standard/ScriptRuntimeImpl.java  2
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectJson.java  9
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/upgrade/AbstractDumpUpgrader.java  1
- modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/cluster/ClusterEventReceiver.java  1
- modules/lib/lib-project/src/main/resources/lib/xp/examples/project/modify.js  1
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/blobstore/AbstractDumpBlobStore.java  3
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/CreateProjectIssuesAccessListCommand.java  3
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ProjectAccessSiteProcessor.java  7
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/command/ApplyProjectReadAccessCommand.java  3
- modules/web/web-session/src/test/java/com/enonic/xp/web/session/impl/HazelcastSessionStoreFactoryActivatorTest.java  1
- modules/lib/lib-project/src/test/resources/test/ModifyProjectPermissionsHandlerTest.js  1
- modules/core/core-hazelcast/src/test/java/com/enonic/xp/core/impl/hazelcast/HazelcastActivatorTest.java  1
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/DefaultFilePaths.java  1
- modules/lib/lib-project/src/test/resources/test/ModifyProjectReadAccessHandlerTest.js  3
- modules/launcher/launcher-impl/src/main/java/com/enonic/xp/launcher/impl/framework/FrameworkService.java  1
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/status/objects/FencedLockObjectReport.java  1
- modules/lib/lib-repo/src/main/resources/lib/xp/examples/repo/modify.js  1
- modules/core/core-project/src/test/java/com/enonic/xp/core/impl/project/ProjectAccessSiteProcessorTest.java  1
- modules/script/script-impl/src/main/java/com/enonic/xp/script/impl/event/ScriptEventManagerImpl.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/RunningTasksReporter.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/ProjectReadAccessType.java  5
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/GetProjectHandler.java  6
- modules/lib/lib-repo/src/main/java/com/enonic/xp/lib/repo/ModifyRepositoryHandler.java  4
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/ListProjectsHandler.java  1
- modules/core/core-repo/src/testFixtures/java/com/enonic/xp/repo/impl/elasticsearch/AbstractElasticsearchIntegrationTest.java  2
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/status/objects/ExecutorServiceObjectReport.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/ProjectResource.java  3
- modules/launcher/launcher-impl/src/main/java/com/enonic/xp/launcher/impl/framework/FrameworkLifecycleAction.java  4
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/AllTasksReporter.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/content/CmsResourceFilter.java  2
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/ModifyProjectReadAccessHandler.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/icon/ProjectIconResource.java  9
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/init/RepoDependentInitializer.java  1
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ProjectAlreadyExistsException.java  1
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/command/AbstractProjectRootCommand.java  1
- modules/jaxrs/jaxrs-impl/src/testFixtures/java/com/enonic/xp/jaxrs/impl/RestRequestBuilder.java  5
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/blobstore/FileDumpBlobStore.java  2
- modules/web/web-jetty/src/main/java/com/enonic/xp/web/jetty/impl/JettyServerDumpReporter.java  1
- modules/lib/lib-project/src/test/java/com/enonic/xp/lib/project/BaseProjectHandlerTest.java  1
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/command/ApplyProjectLanguageCommand.java  1
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/status/objects/MapObjectReport.java  1
- modules/core/core-hazelcast/src/test/java/com/enonic/xp/core/impl/hazelcast/status/objects/HazelcastObjectsReporterTest.java  1
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/NullSessionStoreFactoryActivator.java  1
         

Complexity decreasing per file
==============================
+ modules/core/core-api/src/main/java/com/enonic/xp/init/Initializer.java  -1
+ modules/lib/lib-content/src/main/java/com/enonic/xp/lib/content/BaseContentHandler.java  -1
+ modules/core/core-export/src/main/java/com/enonic/xp/core/impl/export/xml/XsltTransformer.java  -1
+ modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/writer/FileDumpWriter.java  -6
+ modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/index/ReindexExecutor.java  -1
+ modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/rendering/FragmentRenderer.java  -2
+ modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/repository/RepositoryEntryServiceImpl.java  -1
+ modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/resource/ResourceServiceImpl.java  -2
+ modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/reader/FileDumpReader.java  -4
+ modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationFactory.java  -4
+ modules/launcher/launcher-main/src/main/java/com/enonic/xp/launcher/LauncherMain.java  -1
+ modules/lib/lib-node/src/main/java/com/enonic/xp/lib/node/QuerySuggestionParams.java  -2
+ modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/main/MainExecutor.java  -3
+ modules/core/core-cluster/src/test/java/com/enonic/xp/cluster/impl/TestCluster.java  -1
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/RootPermissionsJson.java  -1
+ modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationBuilder.java  -1
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/content/ContentIconUrlResolver.java  -2
         

Clones added
============
- modules/lib/lib-project/src/test/resources/test/CreateProjectHandlerTest.js  2
- modules/core/core-repo/src/testFixtures/java/com/enonic/xp/repo/impl/node/AbstractNodeTest.java  2
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/GetProjectLanguageCommand.java  1
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/CreateProjectRootAccessListCommand.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/ApplyProjectLanguageCommand.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/AbstractProjectRootCommand.java  1
- modules/lib/lib-project/src/test/resources/test/ModifyProjectHandlerTest.js  3
- modules/perftest/perftest-core-repo/src/test/java/com/enonic/xp/repo/bootstrap/PerformanceTestBootstrap.java  2
- modules/lib/lib-project/src/main/resources/lib/xp/project.js  4
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/status/objects/QueueObjectReport.java  1
- modules/lib/lib-project/src/test/resources/test/DeleteProjectHandlerTest.js  3
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/command/GetProjectLanguageCommand.java  1
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/CreateProjectIssuesAccessListCommand.java  1
- modules/lib/lib-project/src/test/resources/test/ModifyProjectPermissionsHandlerTest.js  4
- modules/lib/lib-project/src/test/resources/test/ModifyProjectReadAccessHandlerTest.js  3
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/command/AbstractProjectRootCommand.java  1
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/command/ApplyProjectLanguageCommand.java  1
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/status/objects/MapObjectReport.java  1
         

Clones removed
==============
+ modules/blobstore/blobstore-file/src/main/java/com/enonic/xp/internal/blobstore/file/FileBlobStore.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=6044642&bid=19804798)

File: modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/icon/ProjectIconResource.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+import com.google.common.io.ByteSource;
+
+import com.enonic.xp.admin.impl.rest.resource.content.ResolvedImage;
+import com.enonic.xp.attachment.Attachment;
+import com.enonic.xp.jaxrs.JaxRsComponent;
+import com.enonic.xp.project.Project;
+import com.enonic.xp.project.ProjectName;
+import com.enonic.xp.project.ProjectService;
+import com.enonic.xp.repository.RepositoryService;
```
Comment (jsi) [inline modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/icon/ProjectIconResource.java:24]: Codacy found an issue: [Unused import - com.enonic.xp.repository.RepositoryService.](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=6044642)

File: modules/lib/lib-project/src/main/resources/lib/xp/examples/project/delete.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+// BEGIN
+// Deletes an existing content project inside context with `system.admin` role
+var projectLib2 = require('/lib/xp/project');
+var contextLib = require('/lib/xp/context');
+
+var deleteProject = function () {
+    return projectLib2.delete({
+        id: 'my-project'
+    });
+}
```
Comment (jsi) [inline modules/lib/lib-project/src/main/resources/lib/xp/examples/project/delete.js:18]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=6044642)

File: modules/lib/lib-project/src/main/resources/lib/xp/examples/project/create.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+try {
+    var project = projectLib.create({
+        id: 'my-project',
+        displayName: 'My Content Project',
+        readAccess: {
+            public: true
+        }
+    });
+} catch (e) {
+    log.error('Failed to create a project: ' + e);
```
Comment (jsi) [inline modules/lib/lib-project/src/main/resources/lib/xp/examples/project/create.js:13]: Codacy found an issue: ['log' is not defined.](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=6044642)

File: modules/lib/lib-project/src/main/resources/lib/xp/project.js [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+ * @example
+ * var projectLib = require('/lib/xp/project');
+ *
+ * @module project
+ */
+
+/* global __ */
+
+function required(params, name) {
+    var value = params[name];
```
Comment (jsi) [inline modules/lib/lib-project/src/main/resources/lib/xp/project.js:13]: Codacy found an issue: [Variable Assigned to Object Injection Sink](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=6044642)

File: modules/lib/lib-project/src/main/resources/lib/xp/project.js [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+    if (value === undefined) {
```
Comment (jsi) [inline modules/lib/lib-project/src/main/resources/lib/xp/project.js:14]: Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=6044642)

**Review by rymsha — APPROVED** [review]

**Review by vbradnitski — APPROVED** [review]

**Pull request closed without merging**